include/seqan3/core/algorithm/configuration.hpp

```diff
@@ -561,6 +473,58 @@ class configuration : public std::tuple<configs_t...>
         return make_configuration(std::tuple_cat(head, tail));
     }
     //!\}
+
+    /*!\brief Internal implementation of the get_or interace.
+     *
+     * \tparam this_t The type of this.
+     * \tparam query_t The type of the configuration element to query.
+     * \tparam alternative_t The type of the alternative.
+     *
+     * \param[in] me The perfectly forwarded instance of `*this`.
+     * \param[in] query The queried configuration element [only the type is needed].
+     * \param[in] alternative The alternative configuration element to return if the query_t is not present.
+     *
+     * \details
+     *
+     * Use the type `query_t` to check if such a configuration element is stored in `me`. If this is `true` then
+     * the stored configuration element is returned using perfect forwarding. If this evaluates to `false` the
+     * given alternative is returned. If `query_t` is a class template then it is checked if any
+     * specialisation of this class template is stored.
+     */
+    template <typename this_t, typename query_t, typename alternative_t>
+    static constexpr decltype(auto) get_or_impl(this_t && me,
```

Comment:
why do we need to pass *this as an argument?

Reply:
In order to safe 8 repetitions of the same interface to handle all possible const and reference qualifications and for template and template template arguments.

Reply:
hm ok, I still don't really understand how this works.

Reply:
From your question it was hard to tell, what you asked for. So I try to explain it a little bit more in detail. At first I was calling a member function of this object. This member function is overloaded for `&`, `const &`, `&&` and `const &&`. So all four combinations of const and reference. These 4 versions had to be repeated for the template template version. Now we would need 8 versions of the same function with identical body. So I was outsourcing them into detail functions. But then I had to repeat all 8 versions with the const and reference. But that didn't safe anything. So I made the functions static and perfectly forwarded `*this` to combine all 4 versions in one function call. Now `this` is either an lvalue reference or rvalue reference. In the first case everything is safe since the object lives outside of the function call. The latter case is also safe since the object is not valid any more on the caller site. Using a static function does not bind the calling method to any instance of the object.